magefile.go

```diff
@@ -0,0 +1,96 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+
+	"github.com/mt-sre/devkube/magedeps"
+)
+
+// Directories
+var (
+	// Working directory of the project.
+	workDir string
+	// Cache directory for temporary build files.
+	cacheDir string
+	// Dependency directory.
+	depsDir magedeps.DependencyDirectory
+)
+
+func init() {
+	var err error
+	// Directories
+	workDir, err = os.Getwd()
+	if err != nil {
+		panic(fmt.Errorf("getting work dir: %w", err))
+	}
+
+	depsDir = magedeps.DependencyDirectory(workDir + "/.deps")
+	cacheDir = workDir + "/.cache"
+
+	// Path
+	os.Setenv("PATH", depsDir.Bin()+":"+os.Getenv("PATH"))
+}
+
+// Runs go mod tidy in all go modules of the repository.
+func Tidy() error {
+	if err := sh.Run("go", "mod", "tidy"); err != nil {
+		return fmt.Errorf("tidy main module: %w", err)
+	}
+	return nil
+}
+
+// Testing and Linting
+// -------------------
+
+type Test mg.Namespace
+
+// Runs unittests.
+func (Test) Unit() error {
+	return sh.RunWithV(map[string]string{
+		"CGO_ENABLED": "1",
```

Comment:
Which library requires CGO?

Reply:
No library needs CGO, it's to run the unittests with the race detector `-race`:
https://go.dev/doc/articles/race_detector

```
$ CGO_ENABLED=0 go test -race ./...
go test: -race requires cgo; enable cgo by setting CGO_ENABLED=1
```

Reply:
Interesting I had never seen that.